### PR TITLE
Rename PayPal APIs

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 - Rename PayPal APIs based on #811 
 - Callbacks and result objects still subject to change based on #813 

 ### Checklist

 - [x] Added a changelog entry

### Authors

- @sarahkoop

### CODE REVIEW

File: Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java [hunk 1]
```diff
@@ -46,8 +46,8 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         braintreeClient = getBraintreeClient();
         payPalClient = new PayPalClient(braintreeClient);
         payPalLauncher = new PayPalLauncher(
-                payPalBrowserSwitchResult -> payPalClient.onBrowserSwitchResult(
-                        payPalBrowserSwitchResult, (payPalAccountNonce, error) -> {
+                payPalPaymentAuthResult -> payPalClient.tokenize(
```
Comment (sshropshire) [inline Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java:49]:
```suggestion
                payPalAuthResult -> payPalClient.tokenize(
```
nit: Maybe we can make it more concise by removing the word `payment`?

**Review by sshropshire — APPROVED** [review]

File: Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java (hunk 1 above)
Comment (sarahkoop) [reply Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java]: Do you mean rename the entire class or just the variables in these callbacks? I renamed the variables to `paymentAuthResult` to make it more concise. I think `AuthResult` might be a little too generic for the types but open to opinions on that.

**Review by jaxdesmarais — APPROVED** [review]
🔥

File: Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java (hunk 1 above)
Comment (sshropshire) [reply Demo/src/main/java/com/braintreepayments/demo/PayPalFragment.java]: Yeah I guess the overall naming convention in general to be more specific. I get that adding the word payment may make it more descriptive though. Total take or leave.

I've got the [Swift naming guidelines](https://www.swift.org/documentation/api-design-guidelines/#promote-clear-usage) in mind. I find it to be a useful grammatical guide to naming in any language even though it's geared towards Swift.

File: v5_MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -269,9 +269,9 @@ class MyActivity : FragmentActivity() {
     
     fun onPayPalButtonClick() {
 -       payPalClient.tokenizePayPalAccount(activity, request)
-+       payPalClient.tokenizePayPalAccount(this, request) { payPalResponse, error ->
++       payPalClient.createPaymentAuthRequest(this, request) { payPalPaymentAuthRequest, error ->
 +            error?.let { /* handle error */ }
-+            payPalResponse?.let { 
++            payPalPaymentAuthRequest?.let { 
 +                payPalLauncher.launch(requireActivity(), it)
```
Comment (scannillo) [inline v5_MIGRATION_GUIDE.md:275]:
```suggestion
+                payPalLauncher.launch(requireActivity(), payPalPaymentAuthRequest) 
```

**Pull request merged**